src/main/java/roomescape/infrastructure/JpaReservationRepository.java

```diff
@@ -32,8 +36,13 @@ public List<Reservation> findAll() {
     }
 
     @Override
-    public List<Reservation> findAllWithFilter(Id themeId, Id memberId, LocalDate dateFrom, LocalDate dateTo) {
-        return dao.findAllWithFilter(themeId, memberId, dateFrom, dateTo);
+    public List<Reservation> findAllReservationWithFilter(Id themeId, Id memberId, LocalDate dateFrom, LocalDate dateTo) {
+        return dao.findAllWithFilter(themeId, memberId, dateFrom, dateTo, Status.RESERVED);
+    }
```

Comment:
```suggestion
    public List<Reservation> findAllReservationWithFilter(Id themeId, Id memberId, LocalDate dateFrom, LocalDate dateTo, Status status) {
        return dao.findAllWithFilter(themeId, memberId, dateFrom, dateTo, status);
    }
```
이렇게 status까지 parameter로 입력을 받으면, `findAllWaitingReservations`메서드와의 중복도 줄이고 하나의 메서드를 재사용 할 수 있을 것 같아요.

Reply:
그렇네요! 중복도 방지할 수 있고 무엇보다 status 파라미터를 적절하게 넣어주는 것은 리포지토리보다 서비스의 책임이네요! 수정하였습니다. 감사합니다~